packages/fireworks/src/fireworks-image-model.ts

```diff
@@ -7,21 +7,109 @@ import {
 } from '@ai-sdk/provider-utils';
 
 // https://fireworks.ai/models?type=image
 export type FireworksImageModelId =
   | 'accounts/fireworks/models/flux-1-dev-fp8'
   | 'accounts/fireworks/models/flux-1-schnell-fp8'
+  | 'accounts/fireworks/models/playground-v2-5-1024px-aesthetic'
+  | 'accounts/fireworks/models/japanese-stable-diffusion-xl'
+  | 'accounts/fireworks/models/playground-v2-1024px-aesthetic'
+  | 'accounts/fireworks/models/SSD-1B'
+  | 'accounts/fireworks/models/stable-diffusion-xl-1024-v1-0'
+  | 'accounts/stability/models/sd3-turbo'
+  | 'accounts/stability/models/sd3-medium'
+  | 'accounts/stability/models/sd3'
   | (string & {});
 
+interface FireworksImageModelBackendConfig {
+  urlFormat: 'workflows' | 'image_generation' | 'stability';
+  // Fireworks models use Fireworks API keys and accept a JSON request body.
+  // Stability AI models require a Stability API key, a multipart/form-data
+  // request, and the model specified as `model`: <model-id> in the form data.
+  api: 'fireworks' | 'stability';
+  supportsSize?: boolean;
+}
+
+const modelToBackendConfig: Partial<
+  Record<FireworksImageModelId, FireworksImageModelBackendConfig>
+> = {
+  'accounts/fireworks/models/flux-1-dev-fp8': {
+    urlFormat: 'workflows',
+    api: 'fireworks',
+  },
+  'accounts/fireworks/models/flux-1-schnell-fp8': {
+    urlFormat: 'workflows',
+    api: 'fireworks',
+  },
+  'accounts/fireworks/models/playground-v2-5-1024px-aesthetic': {
+    urlFormat: 'image_generation',
+    api: 'fireworks',
+    supportsSize: true,
+  },
+  'accounts/fireworks/models/japanese-stable-diffusion-xl': {
+    urlFormat: 'image_generation',
+    api: 'fireworks',
+    supportsSize: true,
+  },
+  'accounts/fireworks/models/playground-v2-1024px-aesthetic': {
+    urlFormat: 'image_generation',
+    api: 'fireworks',
+    supportsSize: true,
+  },
+  'accounts/fireworks/models/stable-diffusion-xl-1024-v1-0': {
+    urlFormat: 'image_generation',
+    api: 'fireworks',
+    supportsSize: true,
+  },
+  'accounts/fireworks/models/SSD-1B': {
+    urlFormat: 'image_generation',
+    api: 'fireworks',
+    supportsSize: true,
+  },
+  'accounts/stability/models/sd3-turbo': {
+    urlFormat: 'stability',
+    api: 'stability',
+  },
+  'accounts/stability/models/sd3-medium': {
+    urlFormat: 'stability',
+    api: 'stability',
+  },
+  'accounts/stability/models/sd3': {
+    urlFormat: 'stability',
+    api: 'stability',
+  },
+};
+
+function getUrlForModel(
+  baseUrl: string,
+  modelId: FireworksImageModelId,
+): string {
+  const workflowsUrl = `${baseUrl}/workflows/${modelId}/text_to_image`;
+  switch (modelToBackendConfig[modelId]?.urlFormat) {
+    case 'image_generation':
+      return `${baseUrl}/image_generation/${modelId}`;
+    case 'stability':
+      // For stability models the model id is passed in the form data of the
+      // request rather than as a part of the url.
+      return `https://api.stability.ai/v2beta/stable-image/generate/sd3`;
```

Comment:
Woah - so we go to the stability API directly? This should not be in the fireworks provider then, it should be a separate stability provider. how are fireworks and stability related?

Reply:
Nevermind realized that this is the model name and they do a pass through